panic in checkpoint code

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x58 pc=0x4088cc]

goroutine 18 [running]:
main.appMain.func1(0x0, 0x7efcab814868, 0xc821a2ddd0)
        /go/src/github.com/weaveworks/scope/prog/app.go:56 +0x1c
github.com/weaveworks/scope/vendor/github.com/weaveworks/go-checkpoint.CheckInterval.func1(0x13a52453c000, 0xc820273100, 0x198b590, 0xc82034e540)
        /go/src/github.com/weaveworks/scope/vendor/github.com/weaveworks/go-checkpoint/checkpoint.go:208 +0xe7
created by github.com/weaveworks/scope/vendor/github.com/weaveworks/go-checkpoint.CheckInterval
        /go/src/github.com/weaveworks/scope/vendor/github.com/weaveworks/go-checkpoint/checkpoint.go:213 +0xb7
```

The issue here is that the checkpoint callback can get invoked with a nil first argument and error second argument when something went wrong, yet we pay no attention to the error and immediately dereference the first arg.

--- a/prog/app.go
+++ b/prog/app.go
@@ -53,7 +53,9 @@ func appMain() {
 		Product: "scope-app",
 		Version: app.Version,
 	}, versionCheckPeriod, func(r *checkpoint.CheckResponse, err error) {
-		if r.Outdated {
+		if err != nil {
+			log.Errorf("Error checking version: %v", err)
+		} else if r.Outdated {
 			log.Infof("Scope version %s is available; please update at %s",
 				r.CurrentVersion, r.CurrentDownloadURL)
 		}

